test/TestCase.cpp

```diff
@@ -150,7 +150,9 @@ void EVMVersionRestrictedTestCase::processBytecodeFormatSetting()
 	// EOF only available since Prague
 	solAssert(!eofVersion.has_value() ||solidity::test::CommonOptions::get().evmVersion() >= langutil::EVMVersion::prague());
 
-	std::string bytecodeFormatString = m_reader.stringSetting("bytecodeFormat", "legacy");
+	auto const compileViaYul = m_reader.stringSetting("compileViaYul", "unset");
+	std::string bytecodeFormatString
+		= m_reader.stringSetting("bytecodeFormat", compileViaYul == "false" ? "legacy" : "legacy,>=EOFv1");
```

Comment:
```suggestion
	std::string bytecodeFormatString =
		m_reader.stringSetting("bytecodeFormat", compileViaYul == "false" ? "legacy" : "legacy,>=EOFv1");
```

But I'd generally prefer this:
```suggestion
	std::string bytecodeFormatString = m_reader.stringSetting(
		"bytecodeFormat",
		compileViaYul == "false" ? "legacy" : "legacy,>=EOFv1"
	);
```